waspc/src/Wasp/AppSpec/PackageJson.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Wasp.AppSpec.PackageJson where
+
+import Control.Applicative (liftA2)
+import Data.Aeson.TH
+import Data.Map (Map)
+import qualified Data.Map as M
+import GHC.Generics (Generic)
+import Wasp.AppSpec.App.Dependency (Dependency)
+import qualified Wasp.AppSpec.App.Dependency as D
+
+data PackageJson = PackageJson
+  { _name :: !String,
+    -- todo(filip): do this properly once you merge martin's PR
+    _dependencies :: !(Map String String),
+    _devDependencies :: !(Map String String)
+  }
+  deriving (Show, Generic)
+
+$(deriveJSON defaultOptions {fieldLabelModifier = drop 1} ''PackageJson)
```

Comment:
Oho this is cool, wasn't aware of this!

What is not great here, is that if somebody renames those files to have different prefix (maybe `_` gets removed, or it now has `__`), this will stop working correctly but we will not know. Tests would help, but I don't think you added any (yet)?

Maybe just do it the manual way, so we have type safety? Or you could at least do a bit more involved logic, that will map from "_name" to "name" and so on so it is more visible, or you could check for `_` upfront and drop it only if it exists, ... .

But in the rest of AppSpec data we don't use `_` as prefix, do we? So how about we also not use it here and you rename those additional getters below?

Actually, why are you even storing these as `Map`? Why not as `[Dependency]` if that is what we had before? And if you really do want to store it as you do now, why do you bother with transforming it into `[Dependency]` here, why not do that in the place where `[Dependency]` is needed? Because at this point it seems like AppSpec kind of supports both formats, but that doesn't make much sense to me. I would either go with what you have and remove notion of `Dependency` fromAppSpec, or I would use `Dependency` entirely.

Reply:
Oh ok, sounds good! A TODO here would have helped, as I would be aware of your intentions then + it would increase the likelihood of you not forgetting to do it.

Reply:
True... I guess I was just so happy I finally got it working, I just immediately closed my computer and left :smile: